src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1362,10 +1362,9 @@ void ImportStoreIndirect(TypeDesc type)
 
             var typeVal = StackValue.CreateFromType(type);
             var addressVal = StackValue.CreateFromType(address.Type);
-            if (!value.IsNullReference)
-                CheckIsAssignable(typeVal, addressVal);
 
             CheckIsAssignable(value, typeVal);
+            CheckIsAssignable(value, addressVal);
```

Comment:
Is this going to flag the following as invalid (could you please add this as a test case)?

```
ldarg.0 
ldstr   "a" 
stobj [mscorlib]System.Object
ret
```

I think that the root cause of the problem is a few lines above when we replace null with `GetWellKnownType(WellKnownType.Object)`.

Reply:
I assume arg.0 is still a string&.

No this code would be verified as valid IL by this implementation. Should it be considered invalid? I thought the standard defines that stind.ref is a shorthand for stobj System.Object.

Reply:
The standard says `stind.ref instruction is a shortcut for a stobj instruction that specifies the type pointed at by addr`.

Yes, I think this should be invalid. PEVerify marks it as invalid as well.

Reply:
Oh ok, I see. So `stind.ref` is a shorthand for `stobj pointerType`, therefore there is no check needed whether `typeToken` is assignable to the pointer-type, but stobj does require this check.
You are completely right in this case. The problem does originate from replacing the `typeToken` with `System.Object`.

Reply:
Still, I cannot find where the standard defines that `typeTok` needs to be assignable-to `pointerType`. Checking it does make sense, though.